external/fv3fit/fv3fit/keras/_models/shared/training_loop.py

```diff
@@ -0,0 +1,81 @@
+import tensorflow as tf
+from typing import Iterable, Optional, Sequence, Union, Mapping, Tuple, Callable
+import dataclasses
+import numpy as np
+from fv3fit.keras._models._sequences import _ThreadedSequencePreLoader
+from loaders.batches import shuffle
+import logging
+
+
+# Description of the training loss progression over epochs
```

Comment:
These may need to be imported by any training code that uses `EpochResult`, so I think they should belong here as type hints completing the EpochResult type instead of moved to the dense network code. Otherwise any new model using this history would need to import from the dense code.

Reply:
Ah OK I think that makes sense